tests/rest/media/v1/test_media_storage.py

```diff
@@ -359,7 +359,6 @@ def _test_thumbnail(self, method, expected_body, expected_found):
                 channel.json_body,
                 {
                     "errcode": "M_NOT_FOUND",
-                    "error": "Not found [b'example.com', b'12345?width=32&height=32&method=%s']"
```

Comment:
What caused this to change? It doesn't seem like it should have...

Reply:
this error is built from `request.postpath`. Previously, we were manually setting `postpath` in `make_request` [here](https://github.com/matrix-org/synapse/pull/8761/files#diff-b623bf737404eea7a58674e12e1f83601abfa4a97c3d0da62768ee263a9fbd06L232), based on the full path of the request we are simulating.

we're now leaving `request.process` to populate `postpath`, which it does [here](https://github.com/twisted/twisted/blob/268354f9319d772aba3c565e8b3c390e0440d3c6/src/twisted/web/server.py#L216) based on `request.path`, which is set [here](https://github.com/twisted/twisted/blob/268354f9319d772aba3c565e8b3c390e0440d3c6/src/twisted/web/http.py#L938) and excludes the query-string.

In short: it now better matches the real behaviour.

Reply:
Makes sense! Thanks!